### PR TITLE
fix(tracker): a data row containing `---` is invisible to merge-tracker dedup

### LINKED ISSUE #2265: bug(tracker): a data row whose Notes contain `---` is invisible to merge-tracker dedup (the half #1704 left open)
### Code of Conduct

- [x] I agree to follow the [Code of Conduct](https://github.com/santifer/career-ops/blob/main/CODE_OF_CONDUCT.md)

### Existing issues

- [x] I searched existing issues and this hasn't been reported yet (checked #1704, #1524, #2081, #1965, #1291 — all adjacent, none cover this)

### What happened?

`merge-tracker.mjs` recognizes the markdown separator row with `line.includes('---')`:

```js
// merge-tracker.mjs
for (const line of appLines) {
  if (line.startsWith('|') && !line.includes('---') && !line.includes('Empresa')) {
    const app = parseAppLine(line);
    ...
```

That also matches any **data** row whose free text contains three hyphens — a URL slug such as `Senior-Engineer---Platform-Team`, or an em dash typed as `---`. The sibling `.includes('Empresa')` guard matches a Spanish-market company name the same way.

Such a row never reaches `existingApps`, so **duplicate detection cannot see it**. Re-evaluating that exact company+role appends a second row instead of updating the first in place. The `Existing: N entries` line under-reports too.

**This is the half #1704 deliberately left open.** #1704 fixed the *numbering* side — its separate `usedNumbers` pass reads the `#` cell via a NaN check rather than the heuristic, so a hidden row's number is never reissued. Its comment is explicit that `existingApps` was left alone:

> …can't hide that row's number the way it can hide the row from existingApps (which stays as-is; it drives duplicate detection, not numbering).

So numbering is safe today, but dedup is not.

`verify-pipeline.mjs` shares the heuristic in its row-format check (`if (line.includes('---') || line.includes('Empresa')) continue;`), so a genuinely malformed row carrying three hyphens also escapes the column-count check.

### Steps to reproduce

Tracker with one row whose Notes contain a `---` URL slug:

```
| # | Date | Company | Role | Score | Status | PDF | Report | Notes |
|---|------|---------|------|-------|--------|-----|--------|-------|
| 2 | 2026-01-05 | Acme Corp | Director, Data Platform | 4.3/5 | Evaluated | ❌ | [2](../reports/002-acme-corp-2026-01-05.md) | URL slug still says Senior-Engineer---Platform-Team so title searches miss it. |
| 1 | 2026-01-05 | Globex | Head of Platform | 3.9/5 | SKIP | ❌ | [1](../reports/001-globex-2026-01-05.md) | hybrid NYC. |
```

Re-evaluate that same role at a higher score:

```
2	2026-01-06	Acme Corp	Director, Data Platform	Evaluated	4.8/5	❌	[2](reports/002-acme-corp-2026-01-05.md)	re-evaluated after JD update
```

```bash
node merge-tracker.mjs
```

Output on `main`:

```
📊 Existing: 1 entries, max #2
➕ Add #3: Acme Corp — Director, Data Platform (4.8/5)
```

Two rows for the same company+role. Note the *number* is correct (#3, thanks to #1704) — only dedup failed.

### Expected behavior

```
📊 Existing: 2 entries, max #2
🔄 Update: #2 Acme Corp — Director, Data Platform (4.3→4.8)
```

Row #2 updated in place, no new row.

### Proposed fix

Drop the heuristic from the `existingApps` loop. `parseAppLine` already rejects the header and separator rows — neither has a numeric `#` cell — so the guard is redundant for its intended targets and wrong for data rows. Keep #1704's broader `usedNumbers` pass exactly as-is: it still reserves numbers from rows too malformed for `parseAppLine`, a guarantee the fix must not weaken.

Also worth matching the separator structurally (`/^\|(?:\s*:?-+:?\s*\|)+\s*$/`) where the insert point is located, and in `verify-pipeline`'s row-format check.

Happy to open the PR — I have the fix and a regression test ready (2347 passed, 0 failed).

### CLI tool

Claude Code

### OS and Node.js version

Windows 11, Node v24.18.0

### Output of npm run doctor

`{"onboardingNeeded":false,"missing":[],"warnings":["Playwright MCP tools not detected"],"autoCopied":[]}` — career-ops v1.23.0

### PR DESCRIPTION
**Pull request by LayerC0de**
## What does this PR do?

`merge-tracker.mjs` and `verify-pipeline.mjs` recognized the markdown separator row with `line.includes('---')`, which also matches any **data** row whose free text contains three hyphens (a URL slug like `Senior-Engineer---Platform-Team`, an em dash typed as `---`) — and the sibling `.includes('Empresa')` guard matches a Spanish-market company name. Such a row never reached `existingApps`, so duplicate detection could not see it: re-evaluating that exact company+role appended a second row instead of updating the first in place.

This is the half #1704 deliberately left open. #1704 fixed the *numbering* side via its separate `usedNumbers` pass; its comment is explicit that `existingApps` "stays as-is; it drives duplicate detection, not numbering." Numbering is safe today, dedup is not.

## Related issue

Fixes #2265

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation / translation
- [ ] Refactor (no behavior change)

## Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/santifer/career-ops/blob/main/CONTRIBUTING.md)
- [x] If this is a new feature or architecture change, I opened an issue first (bug fixes, providers, docs & translations are exempt — send those straight in)
- [x] My PR does not include personal data (CV, email, real names, scan results, or pipeline data)
- [x] I ran `node test-all.mjs` and all tests pass
- [x] My changes respect the [Data Contract](https://github.com/santifer/career-ops/blob/main/DATA_CONTRACT.md) (no modifications to user-layer files)
- [x] My changes align with the [project roadmap](https://github.com/santifer/career-ops/discussions/156)

---

### Before / after

Tracker row #2 has a `---` URL slug in its Notes; a TSV re-evaluates that same role at a higher score.

**Before** — appended a duplicate instead of updating:

```
📊 Existing: 1 entries, max #2
➕ Add #3: Acme Corp — Director, Data Platform (4.8/5)
```

**After** — updates in place:

```
📊 Existing: 2 entries, max #2
🔄 Update: #2 Acme Corp — Director, Data Platform (4.3→4.8)
```

### What changed

- **`merge-tracker.mjs`** — drop the heuristic from the `existingApps` loop. `parseAppLine` already rejects the header and separator rows (neither has a numeric `#` cell), so the guard was redundant for its intended targets and wrong for data rows. Also match the separator structurally when locating the insert point.
- **`verify-pipeline.mjs`** — the row-format check skipped any line containing `---`, so a genuinely malformed row carrying three hyphens escaped the column-count check.
- **`tracker-parse.mjs`** — `SEPARATOR_ROW_RE`, `isSeparatorRow`, `isHeaderRow` live here so both readers share one definition and cannot drift, per the module's stated purpose. `isHeaderRow` matches a whole cell reading "company"/"empresa" rather than a substring — matching by substring is what caused this class of bug in the first place.
- **`test-all.mjs`** — 5 assertions: a tracker with one row hidden by `---` and one by "Empresa", both re-evaluated at higher scores, expecting two in-place updates and zero appended rows; plus the `verify-pipeline` short-row case. Fails on `main`, passes here.

### Two deliberate restraints

1. **#1704's `usedNumbers` pass is kept, not merged into the fixed loop.** The two now look redundant, but that pass is deliberately *broader*: it reserves numbers from rows too malformed for `parseAppLine` to return. Collapsing them would silently weaken #1704's guarantee. I updated its comment, since the old wording described a premise this change removes.
2. **No change to dedup semantics** — only to which rows dedup can see. The report-number, exact-num, and fuzzy company+role tiers (including the #1524 req-number guard) are untouched.

`node test-all.mjs`: **2348 passed, 0 failed.**

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed tracker merge deduplication and in-place role/score updates when markdown table separator tokens (`---`) appear in tracker data (e.g., URL slugs).
  * Prevented incorrect handling of company names containing “Empresa” during header/separator detection.
  * Improved detection of the true markdown table header and separator rows and preserved them during merges/validation.
* **Tests**
  * Added regression coverage for the separator/header edge cases, dedup/update behavior, and malformed-row validation failures.

### CODE REVIEW

**Comment by LayerC0de** [comment]
Both review findings addressed — thanks, the second one caught a false claim I'd written.

**1. `isHeaderRow` accepted one header-like cell** (`4a76721`). A malformed data row with an exact `Company`/`Empresa` cell was treated as furniture and skipped row-format validation — the same class of false positive this PR removes, reintroduced one function over. Now keys on the whole schema. Tests: a short row with an exact `Empresa` cell and one with an exact `Company` cell must both be flagged, **and** the genuine header row must still be recognized (a too-strict version would make every tracker report its own header as malformed).

**2. `isHeaderRow` and `detectColumns` had drifted** (`9597eca`). Confirmed, and my comment claiming they "cannot disagree" was wrong — I mirrored the final mapped-schema check but not the literal `company`/`role` pre-filter:

```
ES header   isHeaderRow=true   detectColumns=NULL (falls back to LEGACY_COLMAP)
```

Extracted `headerSchemaMap()`; both now route through it, so disagreement is impossible by construction rather than by assertion. Verified across EN, EN+Location, ES, two header-like malformed rows, a normal data row and the separator — 7/7 agree, and `detectColumns` returns exactly what it did before on each.

**One deliberate omission.** I kept the literal pre-filter inside the shared helper instead of dropping it. Dropping it is the more obviously correct behavior — `tracker-aliases.json` ships `empresa`/`puesto` for exactly the localized case, and #946 explicitly asked for them — but it changes results for every `detectColumns` consumer (`find`, `tracker`, `dedup-tracker`, `set-status`, `normalize-statuses`, `analyze-patterns`, `upskill`, `followup-seed`, `sync-pdf-flags`, `reserve-report-num`), and that isn't something a dedup fix should decide quietly.

Filed separately as #2274, with the concrete consequence: an ES tracker with a Location column reads `Remote` as its score. The fix is one line in the helper this PR adds, so it drops in cleanly afterwards — or say the word and I'll fold it in here instead.

`node test-all.mjs`: **2350 passed, 0 failed.**

**Comment by LayerC0de** [comment]
Folded #2274 in after all (`9a46025`) — flagging the scope change explicitly, since it goes past what the description and my previous comment said.

**What changed.** `detectColumns` no longer gates on literal `company`/`role` cells before consulting `HEADER_ALIASES`, so a fully localized header maps through the aliases `tracker-aliases.json` ships for it. Requiring the full schema is what makes the pre-filter unnecessary: a data row would need five different header labels in five different cells to qualify.

**Why it needed a second file.** `verify-pipeline.mjs` carried its own copy of `LEGACY_COLMAP`, `HEADER_ALIASES` and `detectColumns`. Fixing only the shared module would have left it reading a different layout than `merge-tracker` writes — the drift `tracker-parse.mjs` exists to prevent, and the same half-application #1291 was filed for. It now calls `resolveColumns()`; the triplicate is gone, which is why that file's diff got *smaller*.

**Effect** — an ES tracker with a Location column, the layout #946 was opened about:

```
before → score read as "Remote"
after  → score read as "4.2/5"
```

**Two claims in the PR description are now superseded.** "No change to dedup semantics — only to which rows dedup can see" no longer holds: column detection now resolves layouts it previously fell back to `LEGACY_COLMAP` on, which affects every consumer (`find`, `tracker`, `dedup-tracker`, `set-status`, `normalize-statuses`, `analyze-patterns`, `upskill`, `followup-seed`, `sync-pdf-flags`, `reserve-report-num`). The restraint about keeping #1704's broader `usedNumbers` pass still stands.

Worth being direct about the risk: this only alters behavior for trackers where `detectColumns` previously returned `null` and the caller fell back to fixed positions. For an all-English tracker nothing changes — the pre-filter was already satisfied. For a localized one, callers go from *fixed positions that silently mismatch* to *correct positions*. I believe that is strictly a fix, but it is a behavior change on a code path ten scripts share, so it is entirely reasonable to want it separate — say so and I will split it back out to its own PR against #2274 and rewind this branch to `9597eca`.

Pinned by two tests so it cannot silently revert: the localized header maps with `score@6`, and `isHeaderRow`/`detectColumns` agree 8/8 across EN, EN+Location, ES, ES+Location, two header-like malformed rows, a data row and the separator.

Branch is now 4 commits, 4 files, +288/−44. `node test-all.mjs`: **2352 passed, 0 failed.**

File: tracker-parse.mjs [hunk 1]
```diff
@@ -63,6 +63,68 @@ export function looksLikeScoreCell(v) {
...
+  // FULLY localized header (`| # | Fecha | Empresa | Puesto | … |`) is rejected
+  // here even though HEADER_ALIASES defines `empresa`/`puesto` for exactly that
+  // case, so such a tracker silently falls back to LEGACY_COLMAP. That is a
+  // pre-existing gap, not one this PR introduces, and widening it touches every
+  // detectColumns consumer — so it belongs in its own change.
+  if (!cells.includes('company') || !cells.includes('role')) return null;
+  const map = {};
+  cells.forEach((c, i) => { if (HEADER_ALIASES[c] != null) map[HEADER_ALIASES[c]] = i; });
+  return REQUIRED_HEADER_FIELDS.every(k => map[k] != null) ? map : null;
+}
```
[the remark this answers is not in the document]
Comment (LayerC0de) [reply tracker-parse.mjs:118]: Already done in `9a46025`, pushed ~4 minutes after this review landed — it reviewed `9597eca`, the intermediate commit that still carried the pre-filter. Flagging because GitHub re-anchored the comment to head, so it renders as open.

At head there is no literal `company`/`role` gate anywhere:

```
$ grep -n "includes('company')\|includes('role')" tracker-parse.mjs verify-pipeline.mjs merge-tracker.mjs
(no matches)
```

`headerSchemaMap()` now maps through `HEADER_ALIASES` and validates against `REQUIRED_HEADER_FIELDS` alone. Regression coverage is in as well: a fully localized header maps with `score@6`, and `isHeaderRow`/`detectColumns` agree 8/8 across EN, EN+Location, ES, ES+Location, two header-like malformed rows, a data row and the separator. Filed the underlying bug as #2274 with the concrete symptom (an ES tracker with a Location column read `Remote` as its score); scope change written up at https://github.com/santifer/career-ops/pull/2267#issuecomment-5112176489.

**Comment by santifer** [comment]
Two in a day, @LayerC0de. `line.includes('---')` swallowing any data row with three hyphens in its free text (and `includes('Empresa')` matching Spanish company names) is exactly the half #1704 left open, and your 191 lines of tests pin both the separator and the data-row side. Merged! 🚀
